src/components/ProductTable.jsx

```diff
@@ -32,7 +48,7 @@ const ProductTable = () => {
         defaultChecked={false} 
         onChange={() => {
           // toggle inStockOnly state variable to the opposite of its current value
-          setInStockOnly(!inStockOnly);
+          props.setInStockOnly(!props.inStockOnly);
```

Comment:
Here I would have a function defined in the parent that flips this bool instead of using the state as props here. (I can explain more if that is unclear).

```suggestion
          props.toggleInStock()
```

Reply:
`toggleInStock` would live in ProductPage and look like
```
const toggleInStock = () => {
  setInStockOnly(!inStockOnly);
}
```